server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMDLockManager.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.store.RemoteDirectory;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * A Lock Manager Interface for Remote Store md locks {@code RemoteDirectory}.
+ * @opensearch.internal
+ */
+public abstract class RemoteStoreMDLockManager {
+    private static final Logger logger = LogManager.getLogger(RemoteStoreMDLockManager.class);
+    private final RemoteDirectory lockDirectory;
+    protected static final String SEPARATOR = "__";
```

Comment:
nit: why `__` instead of `_`?

Reply:
using `___` so that it can be used to separate contents from the lock file name.